core/consensus/babe/src/lib.rs

```diff
@@ -1330,37 +1271,38 @@ impl<B, E, Block, I, RA, PRA> BlockImport<Block> for BabeBlockImport<B, E, Block
 				);
 			},
 			(false, true) => {
-				if !check_roots()? {
-					return Err(ConsensusError::ClientImport("Unexpected epoch change".into()));
-				}
+				return Err(ConsensusError::ClientImport("Unexpected epoch change".into()));
 			},
 		}
 
 		// if there's a pending epoch we'll save the previous epoch changes here
 		// this way we can revert it if there's any error
 		let mut old_epoch_changes = None;
 
-		if let Some(next_epoch) = next_epoch_digest {
-			if let Some(enacted_epoch) = enacted_epoch {
-				let enacted_epoch = &enacted_epoch.data;
-				if next_epoch.epoch_index.checked_sub(enacted_epoch.epoch_index) != Some(1) {
-					return Err(ConsensusError::ClientImport(format!(
-						"Invalid BABE epoch change: expected next epoch to be {:?}, got {:?}",
-						enacted_epoch.epoch_index.saturating_add(1),
-						next_epoch.epoch_index,
-					)));
-				}
-			}
+		if let Some(next_epoch_descriptor) = next_epoch_digest {
+			let next_epoch = epoch.increment(next_epoch_descriptor);
 
 			old_epoch_changes = Some(epoch_changes.clone());
 
+			babe_info!("New epoch {} launching at block {} (block slot {} >= start slot {}).",
```

Comment:
As above, should these be downgraded?

Reply:
I'd prefer these to go in the info logs since we won't get them that often (although due to forks we might get a couple ~duplicates at a time).